service/config_provider.go

```diff
@@ -182,14 +183,22 @@ func (cm *configProvider) Shutdown(ctx context.Context) error {
 	return errs
 }
 
+var fileRegexp = regexp.MustCompile("^[A-z]:")
+
 func (cm *configProvider) mergeRetrieve(ctx context.Context) (configmapprovider.Retrieved, error) {
 	var retrs []configmapprovider.Retrieved
 	retCfgMap := config.NewMap()
 	for _, location := range cm.locations {
-		// For backwards compatibility, empty url scheme means "file".
+		// For backwards compatibility:
+		// - empty url scheme means "file".
+		// - "^[A-z]:" also means "file"
 		scheme := "file"
 		if idx := strings.Index(location, ":"); idx != -1 {
-			scheme = location[:idx]
+			if fileRegexp.MatchString(location) {
+				location = scheme + ":" + location
+			} else {
+				scheme = location[:idx]
+			}
```

Comment:
Couple of ideas (independent):
1. Replace everything with a check for ":" only after 2 characters?
2.  Merge the 2 ifs, `if idx := strings.Index(location, ":"); idx != -1 || fileRegexp.MatchString(location) {`

Reply:
I updated the code with number 2

Reply:
+1